pkg/volume/util/hostutil/hostutil_windows.go

```diff
@@ -103,14 +102,6 @@ func isSystemCannotAccessErr(err error) bool {
 func (hu *(HostUtil)) GetFileType(pathname string) (FileType, error) {
 	filetype, err := getFileType(pathname)
 
-	// os.Stat will return a 1920 error (windows.ERROR_CANT_ACCESS_FILE) if we use it on a Unix Socket
```

Comment:
do we need to keep this in place until we drop the symlink compat switch (e.g. merge https://github.com/kubernetes/kubernetes/pull/130008)

same question about whether we have good unit tests for this, and if we can run them on this change with and without https://github.com/kubernetes/kubernetes/pull/130008 layered in

Reply:
I have kept this as we have cherry-picked dropping winsymlink. But I can drop this if it is increasing the complexity and add this in a follow-up PR.

Reply:
is there a unit test showing this now works properly on windows socket files?

Reply:
We had a unit test file testing GetFileType for Windows, it was removed in https://github.com/kubernetes/kubernetes/commit/96e99f564a2e884b6d97510f76340348103e25c0
@marosset should I bring it back?

Reply:
Yes, i think we should bring it back

Reply:
it looks like it just got moved into a common test file tho?
https://github.com/kubernetes/kubernetes/blob/b84cb65f932ae02d06db26c5f9290489e97f9048/pkg/volume/util/hostutil/hostutil_test.go#L50

Reply:
In https://storage.googleapis.com/kubernetes-ci-logs/logs/ci-kubernetes-unit-windows-master/1893027447985672192/artifacts/junit_27.xml we do show the test passing for Windows and I do see a `FileTypeSocket` test case in there

Reply:
that was run against main, let's verify it passed for this PR too

Reply:
k8s.io/kubernetes/pkg/volume/util/hostutil: TestGetFileType

did pass